Webssh updates

rparkinarista opened the pull request:
I can't see how to get copilot to review too @tatanithin how did you do that?

rparkinarista committed:
feat: Add tabbed terminal interface for device SSH access

- Add new /terminal page with tabbed SSH interface
- Add /api/devices endpoint for grouped device list
- Auto-group devices by type (Spine, Leaf, Borderleaf, Host, etc.)
- Add Terminal link to main sidebar navigation
- Include topology diagram toggle overlay

rparkinarista committed:
feat: Add tabbed terminal interface for device SSH access

- Add new /terminal page with tabbed SSH interface
- Add /api/devices endpoint for grouped device list
- Auto-group devices by type (Spine, Leaf, Borderleaf, Host, etc.)
- Add Terminal link to main sidebar navigation
- Include topology diagram toggle overlay

rparkinarista committed:
feat: Add Lab Guides panel and fix terminal layout

- Add Lab Guides button that opens labguides in side panel
- Change layout so side panels don't overlap terminal
- Terminal area shrinks when Lab Guides or Topology is open
- Opening one panel automatically closes the other

rparkinarista committed:
feat: Add collapsible sidebar toggle

- Add toggle button to collapse/expand device sidebar
- Sidebar expanded by default
- Smooth transition animation
- Icon changes between hamburger and arrow based on state

rparkinarista committed:
feat: Auto-focus terminal iframe when switching tabs

- Focus iframe after tab activation for immediate keyboard input

rparkinarista committed:
chore: Rename Terminal link to Switch Access (BETA)

rparkinarista committed:
feat: Add tab reload button and resizable side panels

- Add reload button (↻) on each tab to reload disconnected terminals
- Add draggable resize handle on left edge of Lab Guides and Topology panels
- Panels can be resized between 200px and 80% of terminal area
- Visual feedback when hovering/dragging resize handle

rparkinarista committed:
Revert "feat: Add tab reload button and resizable side panels"

This reverts commit 6544ba63d5955a3df04118ad63e10290632d2a65.

rparkinarista committed:
feat: Add 'All Tabs' dropdown for easy tab navigation

- Shows dropdown button with tab count when tabs are open
- Click to see list of all open tabs with device names and IPs
- Active tab highlighted in dropdown
- Click item to switch to that tab
- Dropdown closes when clicking outside

rparkinarista committed:
feat: Add split view for side-by-side terminals

- Click 'Split View' button to enter split mode
- Left/Right panes each show one terminal
- Click any tab to assign it to the next pane (alternates left/right)
- Each pane shows device name in header
- Click 'Exit Split' to return to single view
- Current terminal automatically placed in left pane on split

rparkinarista committed:
Revert "feat: Add split view for side-by-side terminals"

This reverts commit 042f9b991eec87917f964badbf7c51660c85d085.

rparkinarista committed:
feat(terminal): Add split view for side-by-side terminal comparison

- Add split view mode with separate left/right panes
- Split container uses independent iframes (not moved from tabs)
- Visual indicator shows which pane receives the next device click
- Panes alternate automatically when clicking devices
- Exit Split button cleans up iframes and returns to tab view
- Split toggle in tab bar with active state styling

rparkinarista committed:
fix(terminal): Fix tab bar layout and All Tabs dropdown

- Restructure tab bar with separate scroll area for tabs
- Action buttons (Split View, Lab Guides, Topology) now stay fixed
- Move All Tabs dropdown outside scroll area to prevent clipping
- Tabs scroll horizontally when space is limited
- Buttons no longer move off screen when page shrinks

rparkinarista committed:
fix(terminal): Fix All Tabs dropdown appearing behind iframes

- Increase z-index of dropdown menu to 10000
- Add z-index to tab-bar to establish stacking context above terminal area

rparkinarista committed:
feat(terminal): Add Jump Server link at top of device sidebar

- Add prominent Jump Server link (192.168.0.1) above device groups
- Styled with border and distinct color to stand out
- Shows connected state when terminal is open
- Click to open jump server terminal like any other device

rparkinarista committed:
style(terminal): Match terminal page styling to index.html design system

- Use CSS variables from variables.css (--primary-color, --secondary-color, --neon-green, --navy-blue)
- Apply Proxima Nova font family
- Update sidebar to match index.html left-sidebar styling
- Change accent colors from blue (#4da8da) to yellow (var(--secondary-color))
- Update connected/active states to use neon green
- Match button styles (no border-radius, uppercase, letter-spacing)
- Update tab bar, dropdown, and panel styling
- Use dashed borders for separators like index.html
- Match credentials panel and device list styling

tatanithin commented on nested-labvm/atd-docker/uilanding/src/uilanding.py:
```diff
@@ -883,6 +883,91 @@ def get(self):
+class DevicesAPIHandler(tornado.web.RequestHandler):
+    """API endpoint to return device list grouped by type for terminal page."""
+
+    def get(self):
```
@rparkinarista  I will fix this

tatanithin committed:
Improve terminal accessibility and API security

Adds keyboard accessibility to device items in the terminal UI and refactors tab overflow rendering for better DOM handling. The DevicesAPIHandler now requires authentication, returning a 401 error if the user is not authenticated.
